rir/src/compiler/analysis/refrence_count.h

```diff
@@ -98,8 +98,28 @@ class StaticReferenceCount : public StaticAnalysis<AUses> {
                     count(j);
             }
         };
-
         i->eachArg(apply);
+
+        // Those instructions -- at the rir level -- are allowed to override
+        // inputs, even if the refcount is 1. Therefore if they are used
+        // multiple times, we need to bump the refcount even further.
+        switch (i->tag) {
+        case Tag::Subassign1_1D:
+        case Tag::Subassign2_1D:
+        case Tag::Subassign1_2D:
+        case Tag::Subassign2_2D:
+        case Tag::Inc:
+            if (auto input = Instruction::Cast(i->arg(0).val())) {
+                if (state.uses.count(input) && state.uses.at(input) == 2) {
+                    state.uses[input] = 3;
```

Comment:
maybe rename `uses`, since an instruction will have `uses = 2` if it's used 2+ times but not by a special opcode, and 3+ if it's used 2* by a special opcode.

Also, if `uses = 1` would the special opcode still override? I understand instructions are processed in order, but is this handled:
```r
while (...) {
   use(i)
   i[[n]] <- ...
}
```

Reply:
the static analysis is a fix-point computation, not one pass. so it will take two passes:

first pass it goes from `0 -> 1 -> 2`, then second pass `2 -> 2 -> 3`